src/pages/groupComparison/GroupComparisonMutationMapper.tsx

```diff
@@ -214,4 +357,155 @@ export default class GroupComparisonMutationMapper extends MutationMapper<
+        return (
+            <FisherExactTwoSidedTestLabel
+                dataStore={
+                    this.props.store.dataStore as MutationMapperDataStore
+                }
+                groups={this.props.groups}
+                groupToProfiledPatientCounts={
+                    this.props.groupToProfiledPatientCounts
+                }
+            />
+        );
+    }
+
+    @computed get tableCustomControls(): JSX.Element {
+        return (
+            <div
+                style={{
+                    display: 'flex',
+                    flexDirection: 'row',
+                    alignItems: 'center',
+                    height: 30,
+                    float: 'right',
+                }}
+            >
+                <div style={{ width: 250, marginRight: 7 }}>
+                    <CheckedSelect
+                        name={'groupsSelector'}
+                        placeholder={'Enriched in ...'}
+                        onChange={this.onChange}
+                        options={this.options}
+                        value={this.selectedValues}
+                    />
+                </div>
+                <label className="checkbox-inline" style={{ marginRight: 7 }}>
+                    <input
+                        type="checkbox"
+                        checked={this.significanceFilter}
+                        onClick={this.toggleSignificanceFilter}
+                        data-test="SwapAxes"
+                    />
+                    Significant only
+                </label>
+            </div>
+        );
+    }
+
+    @computed get options(): Option[] {
+        return _.map(this.props.groups, group => {
+            return {
+                label: group.nameWithOrdinal,
+                value: group.nameWithOrdinal,
+            };
+        });
+    }
+
+    @computed get selectedValues() {
+        return this._enrichedGroups.map(id => ({ value: id }));
+    }
+
+    @action.bound
+    onChange(values: { value: string }[]) {
+        this._enrichedGroups = _.map(values, datum => datum.value);
+        onFilterOptionSelect(
+            _.keys(this.rowDataByProteinChangeForEnrichedGroups()),
+            this._enrichedGroups.length === 2 && !this.significanceFilter,
+            this.store.dataStore,
+            DataFilterType.PROTEIN_CHANGE,
+            PROTEIN_CHANGE_FILTER_ID
+        );
+    }
+
+    @action.bound
+    toggleSignificanceFilter() {
+        this.significanceFilter = !this.significanceFilter;
+        onFilterOptionSelect(
+            _.keys(this.rowDataByProteinChangeForEnrichedGroups()),
+            this._enrichedGroups.length === 2 && !this.significanceFilter,
+            this.store.dataStore,
+            DataFilterType.PROTEIN_CHANGE,
+            PROTEIN_CHANGE_FILTER_ID
+        );
+    }
+
+    @action.bound
+    protected resetFilters() {
+        super.resetFilters();
+        this._enrichedGroups = this.props.groups.map(
```

Comment:
again if _enrichedGroups qualifies as a filter then it's name should reflect that